heron/spi/src/java/com/twitter/heron/spi/utils/SchedulerUtils.java

```diff
@@ -436,6 +436,10 @@ public static void persistUpdatedPackingPlan(String topologyName,
                                                SchedulerStateManagerAdaptor stateManager) {
     LOG.log(Level.INFO, "Updating scheduled-resource in packing plan: {0}", topologyName);
     PackingPlanProtoSerializer serializer = new PackingPlanProtoSerializer();
+    if (stateManager.getPackingPlan(topologyName) != null) {
+      stateManager.deletePackingPlan(topologyName);
```

Comment:
This operation works fine in the local filesystem state manager, which is used in the tests, but not the ZK state manager unfortunately.